### PR TITLE
Add more logging around message retention policies support

### PR DESCRIPTION
**Pull request by babolivier**
So we can debug issues like #6683 more easily.

Built on top of #6714 (so I can test it on my HS).

### CODE REVIEW

File: synapse/handlers/pagination.py [hunk 1]
```diff
@@ -130,10 +132,19 @@ def purge_history_for_rooms_in_range(self, min_ms, max_ms):
         else:
             include_null = False
 
+        logger.info(
+            "[purge] Running purge job for range ] %d ; %d ] (include NULLs = %s",
```
Comment (richvdh) [inline synapse/handlers/pagination.py]: is one of these `]`s backwards deliberately?

also, dangling `(`

Comment (babolivier) [reply synapse/handlers/pagination.py]: > is one of these `]`s backwards deliberately?
>

Yep: https://en.wikipedia.org/wiki/Interval_(mathematics)#Including_or_excluding_endpoints

(it's just the notation for intervals that I've always used/seen used)

> also, dangling `(`

Good catch, thanks, fixed in 4087427f0

Comment (richvdh) [reply synapse/handlers/pagination.py]: oh, odd, I've never seen that before; the `(a,b]` is much more familiar.

Comment (babolivier) [reply synapse/handlers/pagination.py]: I guess it's a cultural thing, I had never seen the `(a,b]` one before you mentioned it on a review a few months ago.

**Pull request merged**